File: examples/src/queries/nearest_neighbor_interpolation.cc
#include <wavemap/map/volumetric_data_structure_base.h>
#include <wavemap/utils/query/map_interpolator.h>

#include "wavemap_examples/common.h"

using namespace wavemap;
int main(int, char**) {
  // Create an empty map for illustration purposes
  // NOTE: See the other tutorials on how to load maps from files or ROS topics,
  //       such as the map topic published by the wavemap ROS server.
  VolumetricDataStructureBase::Ptr map;

  // Declare the point to query [in map frame]
  const Point3D query_point = Point3D::Zero();

  // Compute the index that's nearest to the query point
  const FloatingPoint min_cell_width_inv = 1.f / map->getMinCellWidth();
  const Index3D nearest_neighbor_index =
      convert::pointToNearestIndex(query_point, min_cell_width_inv);

  // Query the map
  const FloatingPoint occupancy_log_odds =
      map->getCellValue(nearest_neighbor_index);
  examples::doSomething(occupancy_log_odds);
}

File: examples/python/queries/nearest_neighbor_interpolation.py
import numpy as np
from pywavemap import InterpolationMode
import _dummy_objects

# Load a map
your_map = _dummy_objects.example_map()

# Declare the point to query [in map frame]
query_point = np.array([0.4, 0.5, 0.6])

# Query a single point
occupancy_log_odds = your_map.interpolate(query_point,
                                          InterpolationMode.NEAREST)
print(occupancy_log_odds)

# Vectorized query for a list of points
points = np.random.random(size=(64 * 64 * 32, 3))
points_log_odds = your_map.interpolate(points, InterpolationMode.NEAREST)
print(points_log_odds)
